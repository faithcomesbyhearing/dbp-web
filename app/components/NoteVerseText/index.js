/**
 *
 * NoteVerseText
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import LoadingSpinner from '../LoadingSpinner';

function NoteVerseText({ loading, notePassage }) {
	if (loading) {
		return (
			<div className="verse-text">
				<LoadingSpinner />
			</div>
		);
	}
	return (
		<div className="verse-text">
			&quot;&nbsp;
			{notePassage}
			&nbsp;&quot;
		</div>
	);
}

NoteVerseText.propTypes = {
	notePassage: PropTypes.string.isRequired,
	loading: PropTypes.bool,
};

export default NoteVerseText;
