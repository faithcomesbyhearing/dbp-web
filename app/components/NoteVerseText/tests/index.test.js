import React from 'react';
import renderer from 'react-test-renderer';

import NoteVerseText from '..';

const notePassage = 'Salmon the father of Boaz, whose mother was Rahab. Boaz became the father of Obed, whose mother was Ruth. Obed became the father of Jesse,';

describe('<NoteVerseText />', () => {
  it('Should match snapshot when not loading', () => {
    const tree = renderer.create(<NoteVerseText notePassage={notePassage} loading={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('Should match snapshot when loading', () => {
    const tree = renderer.create(<NoteVerseText notePassage={notePassage} loading />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
