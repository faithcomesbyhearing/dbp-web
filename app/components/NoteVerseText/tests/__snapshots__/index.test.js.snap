// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteVerseText /> Should match snapshot when loading 1`] = `
<DocumentFragment>
  <div
    class="verse-text"
  >
    <div
      class="loading-spinner"
    >
      <span
        class="loading-spinner1 loading-spinner-child"
      />
      <span
        class="loading-spinner2 loading-spinner-child"
      />
      <span
        class="loading-spinner3 loading-spinner-child"
      />
      <span
        class="loading-spinner4 loading-spinner-child"
      />
      <span
        class="loading-spinner5 loading-spinner-child"
      />
      <span
        class="loading-spinner6 loading-spinner-child"
      />
      <span
        class="loading-spinner7 loading-spinner-child"
      />
      <span
        class="loading-spinner8 loading-spinner-child"
      />
      <span
        class="loading-spinner9 loading-spinner-child"
      />
      <span
        class="loading-spinner10 loading-spinner-child"
      />
      <span
        class="loading-spinner11 loading-spinner-child"
      />
      <span
        class="loading-spinner12 loading-spinner-child"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NoteVerseText /> Should match snapshot when not loading 1`] = `
<DocumentFragment>
  <div
    class="verse-text"
  >
    " Salmon the father of Boaz, whose mother was Rahab. Boaz became the father of Obed, whose mother was Ruth. Obed became the father of Jesse, "
  </div>
</DocumentFragment>
`;
